src/MOIWrapper.jl

```diff
@@ -0,0 +1,558 @@
+
+Given a set of linear constraints `l <= a'x <= b` in the model `m`, get the
+constraint primal `a'x` for each constraint, and store in `x`.
+`x` must have one element for each linear constraint.
+"""
+function LQOI.get_linear_primal_solution!(instance::ClpOptimizer, x::Vector{Float64})
+    solution = primal_row_solution(instance.inner)
+    copy!(x,solution)
+end
+
+"""
+get_variable_dual_solution!(m, x::Vector{Float64})
+
+Get the dual solution (reduced-costs) for the variables in the model `m`, and
+store in `x`. `x`must have one element for each variable.
+"""
+function LQOI.get_variable_dual_solution!(instance::ClpOptimizer, x::Vector{Float64})
+    solution = dual_column_solution(instance.inner)
+    copy!(x,solution)
+end
+
+"""
+get_linear_dual_solution!(m, x::Vector{Float64})
+
+Get the dual solution for the linear constraints in the model `m`, and
+store in `x`. `x`must have one element for each linear constraint.
+"""
+function LQOI.get_linear_dual_solution!(instance::ClpOptimizer, x::Vector{Float64})
+    solution = dual_row_solution(instance.inner)
+    copy!(x,solution)
+end
+
+"""
+get_objective_value(m)
+
+Get the objective value of the solved model `m`.
+"""
+function LQOI.get_objective_value(instance::ClpOptimizer)
+    return objective_value(instance.inner)
+end
+
+"""
+get_termination_status(m)
+
+Get the termination status of the model `m`.
+"""
+function LQOI.get_termination_status(instance::ClpOptimizer)
+    s = ClpCInterface.status(instance.inner)
+    if s == 0 
+        return MOI.Success
+    elseif s == 1
+        return MOI.InfeasibleNoResult
+    elseif s == 2
+        return MOI.UnboundedNoResult
+    # if s in [0, 1, 2] 
+    #     return MOI.Success
+    elseif s == 3
+        return MOI.OtherLimit
+    elseif s == 4
+        return MOI.OtherError
+    else
+        error("status returned by Clp must be in [0,1,2,3,4]")
+    end
+end
+
+"""
+get_primal_status(m)
+
+Get the primal status of the model `m`.
+"""
+function LQOI.get_primal_status(instance::ClpOptimizer)
+    if primal_feasible(instance.inner)
+        return MOI.FeasiblePoint
+    else    
+        return MOI.UnknownResultStatus
+    end
+end
+
+"""
+get_dual_status(m)
+
+Get the dual status of the model `m`.
+"""
+function LQOI.get_dual_status(instance::ClpOptimizer)
+    if dual_feasible(instance.inner)
+        return MOI.FeasiblePoint     
+    else
+        return MOI.UnknownResultStatus
+    end    
+end
+
+"""
+get_number_variables(m)::Int
+
+Get the number of variables in the model `m`.
+"""
+function LQOI.get_number_variables(instance::ClpOptimizer)
+    return get_num_cols(instance.inner)
+end
+
+"""
+add_variables!(m, n::Int)::Void
+
+Add `n` new variables to the model `m`.
+"""
+function LQOI.add_variables!(instance::ClpOptimizer, n::Int)::Void
+    for i in 1:n
+        numberInColumn = 0
+        rows = Int32[]
+        elements = Float64[]
+        add_column(instance.inner, numberInColumn,  rows, elements, -Inf, +Inf, 0.0)
```

Comment:
So `Inf` bounds are okay here? Not `1e20`?

Reply:
This can probably just be `add_column(instance.inner, 0, Int32[], Float64[], -Inf, Inf, 0.0)`